pkg/controller/applicationintegration/sqs/contoller.go

```diff
@@ -0,0 +1,244 @@
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sqs
+
+import (
+	"context"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	awssqs "github.com/aws/aws-sdk-go-v2/service/sqs"
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/crossplane/crossplane-runtime/pkg/meta"
+	"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+
+	"github.com/crossplane/provider-aws/apis/applicationintegration/v1alpha1"
+	awsv1alpha3 "github.com/crossplane/provider-aws/apis/v1alpha3"
+	awsclients "github.com/crossplane/provider-aws/pkg/clients"
+	"github.com/crossplane/provider-aws/pkg/clients/sqs"
+)
+
+const (
+	errNotQueue                 = "managed resource is not a Queue custom resource"
+	errKubeUpdateFailed         = "cannot update Queue custom resource"
+	errQueueClient              = "cannot create Queue client"
+	errGetProvider              = "cannot get provider"
+	errGetProviderSecret        = "cannot get provider secret"
+	errCreateFailed             = "cannot create Queue"
+	errInvalidNameForFifoQueue  = "cannot create Queue, FIFO queue name must have .fifo suffix"
+	errDeleteFailed             = "cannot delete Queue"
+	errGetQueueAttributesFailed = "cannot get Queue attributes"
+	errListQueueTagsFailed      = "cannot list Queue tags"
+	errUpdateFailed             = "failed to update the Queue resource"
+	errStatusUpdate             = "cannot update status"
+	fifoQueueSuffix             = ".fifo"
+)
+
+type connector struct {
+	kube        client.Client
+	newClientFn func(ctx context.Context, credentials []byte, region string, auth awsclients.AuthMethod) (sqs.Client, error)
+}
+
+type external struct {
+	client sqs.Client
+	kube   client.Client
+}
+
+// SetupQueue adds a controller that reconciles Queue.
+func SetupQueue(mgr ctrl.Manager, l logging.Logger) error {
+	name := managed.ControllerName(v1alpha1.QueueGroupKind)
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1alpha1.Queue{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha1.QueueGroupVersionKind),
+			managed.WithExternalConnecter(&connector{kube: mgr.GetClient(), newClientFn: sqs.NewClient}),
+			managed.WithInitializers(),
```

Comment:
It appears you are overriding initializers, but also using `ExternalName` in making calls. I think you will likely want remove this line or include `NewNameAsExternalName` and any other initializers you would need, ref: https://github.com/crossplane/provider-aws/blob/276140071bc332832593952092bcb2507af56f23/pkg/controller/database/rdsinstance.go#L73

Reply:
The `ExternalName` is being set in the `Create` method, with the value of `QueueURL`. That is being done to not depend on Status to maintain `QueueURL`, which is the identifier for the queue.